core/src/main/java/de/danoeh/antennapod/core/util/FeedItemUtil.java

```diff
@@ -75,4 +76,21 @@ public static boolean containsAnyId(List<FeedItem> items, long[] ids) {
         return false;
     }
 
+    /**
+     * Get the link for the feed item for the purpose of Share. It fallbacks to
+     * use the feed's link if the named feed item has no link.
+     */
+    public static String getLinkWithFallback(FeedItem item) {
+	    String link = null;
+	    if (item != null) {
+            link = item.getLink();
+            if (link == null) {
+                Feed feed = item.getFeed();
+                if (feed != null) {
+                    link = feed.getLink();
+                }
+            }
+        } // else null item, can only return null
+        return link;
```

Comment:
I think it is easier to read if you don't use nested `if`s. Same for the other method.

```
if (item == null) {
    return null;
} else if (item.getLink() != null) {
    return item.getLink();
} else if (item.getFeed() != null) {
    return item.getFeed().getLink();
}

return null;
```

Reply:
I've pushed a new check-in flattening the nested `if`s.  

I am somewhat weary of many `return`s, but I do agree it's better than nested `if`s in this case.